### PR TITLE
Change IR logo

### PR DESCRIPTION
**Pull request by nbiton**
Stuff that already works:
* Marking appears over default logo
* Clicking the marking makes the "add logo" button appear.
* The dialog to change the logo appears when clicking that button
* The dialog allows for logo selection and preview.

Actually changing the logo on the report is still WIP.

This PR is based off #636 and shouldn't be merged until it's merged

### CODE REVIEW

**Comment by nbiton** [comment]
screenshot from 2018-11-16 22-33-57
screenshot from 2018-11-16 22-34-12
screenshot from 2018-11-16 22-34-35
screenshot from 2018-11-16 22-37-52
screenshot from 2018-11-16 22-38-00
screenshot from 2018-11-16 22-38-17
screenshot from 2018-11-16 22-42-12

This PR is done from a functional standpoint, but a single issue remains: the marking around the logo (dashed stroke rectangle) remains fixed and doesn't scroll/disappear with the rest of the report preview.

**Comment by nbiton** [comment]
To enable this for a user, it is currently required to set "premiumFeaturesEnabled: true" directly on that user's document in Mongo

**Comment by nbiton** [comment]
Although this is now completely done (pending function review @franck-boullier), the dashed rectangular marking would still appear fixed on the PR app, due to the fact the iframe's contents (the IR preview html) is not coming from the same domain as the PR app. This should work as expected on the dev env after merge, as the preview htmls' and the dev UI's domains are the same.

**Comment by franck-boullier** [comment]
This does not seem to work as expected...

# The test:

Test done on Galaxy S8 and on FireFox on a DT

- log in to the heroku app as leonel PASSED
- create an [Inspection Report](https://unee-t-pr-637.herokuapp.com/report/557/draft) PASSED
- Create some remarks and comment PASSED
- Add a picture PASSED
- Go to '[Preview](https://unee-t-pr-637.herokuapp.com/report/557/preview)' PASSED
- Try to change the logo - FAILED

## Expected result:

The existing Unee-T logo is clickable as per the design
image


## Actual result

The logo is not clickable
image

**Comment by nbiton** [comment]
> To enable this for a user, it is currently required to set "premiumFeaturesEnabled: true" directly on that user's document in Mongo

@franck-boullier did you do that bit? it's off by default currently

**Comment by franck-boullier** [comment]
> set "premiumFeaturesEnabled: true" directly on that user's document in Mongo

Oops: no I did not do this...
I'll try that.

**Comment by franck-boullier** [comment]
> I'll try that.

@nbiton I just realized that I have no access to the Mongo associated to the Heroku App. Can you turn this on for the user leonel please?

**Comment by nbiton** [comment]
I just activated it for leonel

**Comment by franck-boullier** [comment]
I'd suggest that we rename the attribute in the user document in Mongo from "premiumFeaturesEnabled" to "canChangeIRLogo" ---> this should give us much more flexibility/clarity in the future.

It'll be difficult to understand what "premiumFeaturesEnabled" does

For instance we will have users in the Enterprise plan who are allowed to change the logo of the IR and users in the Enterprise plan who are NOT allowed to change the IR logo, we need to keep that I  mind

**Comment by franck-boullier** [comment]
# The tests:

Test done on Galaxy S8 and on FireFox on a DT

- log in to the heroku app as leonel PASSED
- create an [Inspection Report](https://unee-t-pr-637.herokuapp.com/report/557/draft) PASSED
- Create some remarks and comment PASSED
- Add a picture PASSED
- Go to '[Preview](https://unee-t-pr-637.herokuapp.com/report/557/preview)' PASSED
- Try to change the logo - PASSED
- Close the inspection report preview - PASSED
- Open a different IR for the same user and check that the new logo is displayed - PASSED.
- Finalize an IR - check that the IR is the new logo - PASSED
- Go to a different report - change the logo again - PASSED
- Check that the logo for the previsously finalized IR is still the old logo - PASSED

So far all tests are OK

# Missing functionality and suggested improvements:

- There should be a functionality to 'Reset logo to default' where the user can revert back to the default Unee-T logo
- We should add a message with the recommended size (in px) for the logo to make sure user know which format they need to upload.
- If the logo does not meet the required format, we should allow the user to crop/re-size the logo to fit the format of the placeholder in the IR.
- Rename the attribute in the user document in Mongo from "premiumFeaturesEnabled" to "canChangeIRLogo" (or something more specific.

**Comment by franck-boullier** [comment]
> There should be a functionality to 'Reset logo to default' where the user can revert back to the default Unee-T logo
> Shouldn't be a problem, but we need a design for that

Can't we go with a super simple button (or even a link) somewhere here?
image

> Any size should be fine really (...)

I've tested that and this not a correct assumption: when the logo is auto stretched it's rapidly very ugly, even with a simple square logo.

Also we want to avoid the scenario where a user tries to upload a 4,000x4,000 px picture of himself that will take ages to upload (I tried that and that is very slow...)

---> Quick fix to mitigate these 2 risks would be to give the recommended size in px for the logo.

> Rename the attribute in the user document in Mongo from "premiumFeaturesEnabled" 
> This doesn't really matter much as it's temporary

Hmmm... @nbiton this is based on several assumptions which are risky IMO:
- The "temporary" will only last for a short while ---> we are notoriously bad at predicting how fast we can implement new stuff.
- You will be the one that will do the change ---> hopefully we'll have more resources soon and it'll be better is the description of what this variable controls is precise rather than too vague or potentially misleading.
- We will remember this conversation when it's time to update the code ---> I'm about 99% sure that I will not remember

**Comment by nbiton** [comment]
I renamed the attribute used as a switch for this feature to "customReportLogoEnabled" and added an automatic migration to flag it as "true" for all existing users.

Pending issues:
* "Reset logo to default" - we need to decide where that would show up and how it's going to look like.
* Recommended format/size - we need to decide what criteria we want to highlight, and how that would appear to the user

**Comment by nbiton** [comment]
@kiatlim @michkwt do you have any suggestions regarding the two pending issues?

**Comment by kiatlim** [comment]
@nbiton recommended format / size and reset logo to default should appear in the model window where we upload the image. Let me create a design for that and upload it for preview now.

**Comment by franck-boullier** [comment]
@nbiton 
About the pending issues:

- Recommended format/size - we need to decide what criteria we want to highlight, and how that would appear to the user ---> this will be taken care of in another PR

- "Reset logo to default" - we need to decide where that would show up and how it's going to look like. Once we have the design how long do you think it'll take to implement this?
My idea is that if it takes more than 1 day then we do this in another PR too.

**Comment by yvonnechia** [comment]
Hi @franck-boullier @nbiton as discussed earlier, here's the revised visual flow:

add logo flow with reset

**Comment by nbiton** [comment]
I think the "Reset to default" that's in the 3rd screen is misplaced, as it doesn't really serve the same function you see in the 4th screen. 
The 3rd screen is only showing a preview of the custom logo **before** it was confirmed and saved. There's no need to "Reset to default" a preview.

Also, I suggest rewording "Upload another" to "Replace", as "Upload another" suggests you can upload multiple files to be used simultaneously.

**Comment by yvonnechia** [comment]
> I think the "Reset to default" that's in the 3rd screen is misplaced, as it doesn't really serve the same function you see in the 4th screen.
> The 3rd screen is only showing a preview of the custom logo **before** it was confirmed and saved. There's no need to "Reset to default" a preview.
> 
> Also, I suggest rewording "Upload another" to "Replace", as "Upload another" suggests you can upload multiple files to be used simultaneously.

Yes that makes sense. Thanks! Made some tweaks accordingly. 

add logo flow with reset 2

**Comment by nbiton** [comment]
@yvonnechia It looks great, thanks.

Are you sure there's no need for a confirmation dialog after "Reset to default" is clicked?

**Comment by nbiton** [comment]
screenshot from 2018-11-27 16-49-58
screenshot from 2018-11-27 16-50-16
screenshot from 2018-11-27 16-50-51
screenshot from 2018-11-27 16-51-09

@franck-boullier 
All done.
Once tested/confirmed, I'll fix the merge conflicts and merge. I'm not fixing them now as it is likely to break the PR build, but necessary for merging to master (where the build is expected to be fine).
